docs/src/Dropdown.doc.js

```diff
@@ -0,0 +1,642 @@
+        const [open, setOpen] = React.useState(false);
+        const [selected, setSelected] = React.useState(null);
+        const anchorRef = React.useRef(null);
+        const handleSelect = ({item}) => {
+          setSelected(item);
+        };
+
+        return (
+          <Box display="flex" justifyContent="center">
+            <Button
+              accessibilityHaspopup
+              accessibilityExpanded={open}
+              iconEnd="arrow-down"
+              text="Menu"
+              inline
+              ref={anchorRef}
+              selected={open}
+              onClick={ () => setOpen((prevVal) => !prevVal) }
+            />
+            {open && (
+              <Dropdown onSelect={(event, item) => {console.log("selectingg", event, item);}} anchor={anchorRef.current} onDismiss={() => {setOpen(false)}}>
+                <Dropdown.Item
+                index={20}
+                  handleSelect={handleSelect}
+                  selected={selected}
+                  option={{ value: 'item 1', label: 'Item 1' }}
+                />
+                <Dropdown.Item
+                  handleSelect={handleSelect}
+                  selected={selected}
+                  option={{ value: 'item 2', label: 'Item 2 with a really long, detailed, complex name' }}
+                />
+                <Dropdown.Item
+                  isExternal
+                  href="https://pinterest.com"
+                  option={{ value: 'item 3', label: 'Item 3 with a really long, detailed, complex name' }}
+                />
+                <Dropdown.Item
+                  handleSelect={handleSelect}
+                  selected={selected}
+                  badgeText="New"
+                  option={{ value: 'item 4', label: 'Item 4' }}
+                />
+                <Dropdown.Item
+                  isExternal
+                  badgeText="New"
+                  option={{ value: 'item 5', label: 'Item 5 with a really long, detailed name' }}
+                  href="https://pinterest.com"
+                />
+                <Dropdown.Item
+                  option={{ value: 'item 6', label: 'Item 6 navigates internally' }}
+                  href="/typeahead"
+                />
+              </Dropdown>
+            )}
+          </Box>
+        );
+      }`}
+  />
+);
+
+card(
+  <PropTable
+    Component={Dropdown}
+    name="Dropdown"
+    id="Dropdown"
+    props={[
+      {
+        name: 'anchor',
+        type: '?HTMLElement',
+        description: 'Ref for the element that the Dropdown will attach to',
+        href: 'default',
+      },
+      {
+        name: 'children',
+        required: true,
+        type:
+          'ChildrenArray<Element<typeof DropdownItem | typeof DropdownSection>>',
+        description:
+          'Should consist of Dropdown.Item and/or Dropdown.Section components',
+        href: 'default',
+      },
+      {
+        name: 'headerContent',
+        type: 'React.Node',
```

Comment:
I think we're actually ok with allowing anything here, since there are already varying uses of this header in the product that we want to account for.